pkg/api/pss.go

```diff
@@ -52,6 +49,23 @@ func (s *server) pssPostHandler(w http.ResponseWriter, r *http.Request) {
 		targets = append(targets, target)
 	}
 
+	recipientVar := mux.Vars(r)["recipient"]
+	var recipient *ecdsa.PublicKey
+	if recipientVar == "" {
+		// use topic-based encryption
+		privkey := crypto.Secp256k1PrivateKeyFromBytes(topic[:])
+		recipient = &privkey.PublicKey
+	} else {
+		var err error
+		recipient, err = pss.ParseRecipient(recipientVar)
+		if err != nil {
+			s.Logger.Debugf("pss recipient: %v", err)
+			s.Logger.Error("pss recipient")
+			jsonhttp.InternalServerError(w, nil)
```

Comment:
BadRequest

Reply:
That is pre-existing code, I will change it.